rasa/cli/run.py

```diff
@@ -77,26 +77,57 @@ def run(args: argparse.Namespace):
 
     args.model = _validate_model_path(args.model, "model", DEFAULT_MODELS_PATH)
 
-    if not args.enable_api:
-        # if the API is enabled you can start without a model as you can train a
-        # model via the API once the server is up and running
-        from rasa.model import get_model
-
-        try:
-            get_model(args.model)
-        except ModelNotFound:
-            print_error(
-                "No model found. Train a model before running the "
-                "server using `rasa train` and use '--model' to provide "
-                "the model path."
-            )
-            return
-
     args.endpoints = get_validated_path(
         args.endpoints, "endpoints", DEFAULT_ENDPOINTS_PATH, True
     )
     args.credentials = get_validated_path(
         args.credentials, "credentials", DEFAULT_CREDENTIALS_PATH, True
     )
 
-    rasa.run(**vars(args))
+    if args.enable_api:
+        rasa.run(**vars(args))
+
+    # if the API is not enable you cannot start without a model
+    # make sure either a model server, a remote storage, or a local model is
+    # configured
+
+    from rasa.model import get_model
+    from rasa.core.utils import AvailableEndpoints
+
+    # start server if remote storage is configured
+    if args.remote_storage is not None:
+        rasa.run(**vars(args))
+
+    # start server if model server is configured
+    try:
+        endpoints = AvailableEndpoints.read_endpoints(args.endpoints)
+        model_server = endpoints.model if endpoints and endpoints.model else None
+        model_server_set = model_server is not None
+    except ValueError:
+        model_server_set = False
```

Comment:
If for some reason we don't get a `ValueError` but rather another error, will we not have a value for `model_server_set` leading to an undefined variable on L109?

Reply:
or I suppose that Error would throw and we wouldnt make it to 109 anyway would we 😄

Reply:
yeah, if another exception will be thrown, then the server would not start at all. we could also just remove the try catch and throw the exception, as we also do not wrap the exception later. so no matter what we do here, an exception will be thrown at some point, if the endponits.yml file is invlaid.

Reply:
sweet 👍